framework/tests/all-systems/Issue3598.java

```diff
@@ -0,0 +1,25 @@
+import java.util.function.Function;
+
+class Issue3598 {
+
+    static class DClass extends EClass {}
+
+    static class EClass<F> {}
+
+    // Must be Function, can't use interface defined in this class.
```

Comment:
@smillst There isn't an interface defined in this class.  Could you clarify this comment?